imports/plugins/core/orders/client/containers/orderDashboardContainer.js

```diff
@@ -194,6 +260,397 @@ class OrderDashboardContainer extends Component {
+        showCancelButton: true,
+        showCloseButton: true,
+        allowOutsideClick: false,
+        confirmButtonText: i18next.t("order.approveBulkOrderAction"),
+        cancelButtonText: i18next.t("order.cancelBulkOrderAction")
+      }, (setSelected) => {
+        if (setSelected) {
+          // set status of order(s) if this action is confirmed
+          this.shippingStatusUpdateCall(selectedOrders, status);
+        }
+      });
+      // if the order(s) are following proper flow, set the status
+    } else if (!falsePreviousStatuses && falseCurrentState) {
+      this.shippingStatusUpdateCall(selectedOrders, status);
+      // display alert if order(s) are already in this state
+    } else if (!falsePreviousStatuses && !falseCurrentState && trueCurrentState) {
+      Alerts.alert({
+        text: i18next.t("order.orderAlreadyInState", {
+          orderText: this.displayOrderText(selectedOrders),
+          status: status
+        })
+      });
+    }
+  }
+
+  displayRegressionAlert = (
+    selectedOrders, ordersToRegress, status, whichFalseState, falsePreviousStatuses, falseCurrentState,
+    trueCurrentState) => {
+    const capitalizeStatus = status[0].toUpperCase() + status.substr(1).toLowerCase();
+    let orderText = "";
+
+    if (ordersToRegress > 1) {
+      orderText = "orders";
+    } else {
+      orderText = "order";
+    }
+
+    Alerts.alert({
+      text: i18next.t("order.bulkOrdersRegressionAlert", {
+        ordersToRegress: ordersToRegress, orderText: orderText, status: capitalizeStatus
+      }),
+      type: "warning",
+      showCancelButton: true,
+      showCloseButton: true,
+      allowOutsideClick: false,
+      confirmButtonText: i18next.t("order.approveBulkOrderActionRegression"),
+      cancelButtonText: i18next.t("order.cancelBulkOrderAction")
+    }, (regress) => {
+      if (regress) {
+        // if some of the order(s) want to skip the previous state, display warning alert for skipping states
+        if (falsePreviousStatuses) {
+          this.displayAlert(
+            selectedOrders, status, whichFalseState, falsePreviousStatuses, falseCurrentState, trueCurrentState
+          );
+        } else {
+          // set status of order(s) if this action is confirmed
+          this.shippingStatusUpdateCall(selectedOrders, status);
+        }
+      }
+    });
+  }
+
+  pickedShippingStatus = (selectedOrders, status) => {
+    let isNotPicked = 0;
+    let isPicked = 0;
+    let ordersToRegress = 0;
+
+    selectedOrders.forEach((order) => {
+      // check if the order(s) are in this state already or in the previous state
+      if (order.shipping[0].workflow.status === "new") {
+        isNotPicked++;
+      } else if (order.shipping[0].workflow.status === "coreOrderWorkflow/picked") {
+        isPicked++;
+      } else {
+        // check if the selected order(s) are being regressed back to this state
+        if (order.shipping[0].workflow.workflow.includes("coreOrderWorkflow/picked")) {
```

Comment:
Would like to have more comments explaining the different directions these `if..else if.. else` blocks can take.